pgxn/neon/pagestore_smgr.c

```diff
@@ -332,19 +365,22 @@ prefetch_read(PrefetchRequest *slot)
 void
 prefetch_on_ps_disconnect(void)
 {
-	for (; MyPState->ring_receive < MyPState->ring_unused; MyPState->ring_receive++)
+	MyPState->ring_flush = MyPState->ring_unused;
+	for (; MyPState->ring_receive < MyPState->ring_unused;)
```

Comment:
it seems that it can be replaced with `while` cycle because you've changed it a bit